1_fetch/src/get_sb_data.R

```diff
@@ -206,18 +72,95 @@ download_children <-function(sites, sb_table_reduced, dldir, workdir, outdir, ou
+    log_item <- tibble(sb_id, last_update)
+    sb_log <- bind_rows(sb_log, log_item)
+  }
+  
+  write_csv(sb_log, file_out)
+  return(file_out)
+  
+}
+
+prep_feature_vars <- function(sb_var_data, sites) {
+  
+  data <- sites %>%
+    select(COMID, LAT, LON, npdes, fwwd, strg, devl, cndp)
+  
+  for (i in 1:length(sb_var_data)) {
+    filepath <- sb_var_data[[i]][1]
+    data_temp <- read_csv(filepath, show_col_types = FALSE) %>%
+      group_by(COMID) %>%
+      summarise_all(mean)
+    data <- left_join(data, data_temp, by = "COMID")
+  }
+  
+  data$npdes <- as.numeric(data$npdes)
+  data$fwwd <- as.numeric(data$fwwd)
+  data$strg <- as.numeric(data$strg)
+  data$devl <- as.numeric(data$devl)
+  data$cndp <- as.numeric(data$cndp)
+  
+  data <- data %>% 
+    group_by(COMID) %>%
+    summarise_all(mean)
+  
+  land_cover <- data %>%
+    select(COMID, 
+           contains("SOHL40"), contains("SOHL50"), contains("SOHL60"),
+           contains("SOHL70"), contains("SOHL80"), contains("SOHL90"), 
+           contains("SOHL00")) %>%
+    pivot_longer(!COMID, names_to = "name", values_to = "value") %>%
+    mutate(unit = str_sub(name, 1, 3), 
+           year = if_else(str_sub(name, 9, 10) == "00", 2000, 
+                          as.numeric(paste0("19", str_sub(name, 9, 10)))), 
+           class = as.numeric(str_sub(name, 12))) %>%
+    group_by(COMID, unit, year, class) %>%
+    summarise(value = mean(value), .groups = "drop") %>%
+    mutate(new_class = case_when(class == 1 ~ "WATER", class == 2 ~ "DEVELOPED", 
+                                 class == 3 ~ "FOREST", class == 4 ~ "FOREST", 
+                                 class == 5 ~ "FOREST", class == 6 ~ "MINING", 
+                                 class == 7 ~ "BARREN", class == 8 ~ "FOREST", 
+                                 class == 9 ~ "FOREST", class == 10 ~ "FOREST", 
+                                 class == 11 ~ "GRASSLAND", class == 12 ~ "SHRUBLAND", 
+                                 class == 13 ~ "CROPLAND", class == 14 ~ "HAY/PASTURE", 
+                                 class == 15 ~ "WETLAND", class == 16 ~ "WETLAND", 
+                                 class == 17 ~ "ICE/SNOW")) %>%
+    group_by(COMID, unit, year, new_class) %>%
+    summarise(value = sum(value), .groups = "drop")
+  
+  comid_out_conus <- land_cover %>%
+    group_by(COMID, unit, year) %>%
+    summarise(lc_sum = sum(value), .groups = "drop") %>%
+    filter(lc_sum < 90 | lc_sum > 101)
+  comid_out_conus <- unique(comid_out_conus$COMID)
+  data <- data %>%
+    subset(!(COMID %in% comid_out_conus))
+  
+  land_cover <- land_cover %>%
+    group_by(COMID, unit, new_class) %>%
+    summarise(value = mean(value), .groups = "drop") %>%
+    mutate(label = paste0(unit, "_SOHL_", new_class)) %>%
+    select(COMID, label, value) %>%
+    pivot_wider(names_from = "label", values_from = "value")
+  land_cover <- land_cover %>%
+    subset(!(COMID %in% comid_out_conus))
```

Comment:
Did you rescale the proportions to 100 after filtering? I don't see that step. I think it would go here. You can apply the rescaling to all land cover columns because they all do not sum exactly to 100.

Reply:
I forgot - good call. Will add in a scaling step. 

Question for self - do other feature variables require rescaling as well??